### DISCUSSION TITLE
Albums not show in view "Album"

### DISCUSSION THREAD

**Discussion by pdo59** [rern/rAudio#71] [Fixed Issues]
Hello,
your software is very interesting! I really like it. But there are some things I don't understand.
I have about 6500 albums in my collection; but after updating the library, in the view "Album" I see only 56 albums. I believe that they are only those related to wav files. How Is it possible to view all the album in the library? 
Another question: why in the "AlbumArtist" view is not showed the thumbnail?
I have installed the latest version.

**Comment by ceejayemm** [comment]
Hi

Raudio does indeed 'find' .wav files in your sources automatically. To include other audio files in your Library you need to manually update the library. To do this click the small icon to the right of the Library item in the Setup (cog) menu.

You might also want to be aware of issue #12 which also affects me with RPi Zero and Bluetooth.

Chris

**Reply by pdo59** [reply]
Hi Chris, 
yes, that's what I did. I first tried to rebuild the whole library, then, having no useful results, I tried a small part of my library. The results are the same: in "/ srv / httpd / data / mpd" the files "album" and "albumwav" are empty, instead the files "albumartist", "albumbyartist", "artist", "composer", "date “,”genre" contain the correct information.
I’m tried more times... without success

<img width="917" alt="Schermata 2021-01-04 alle 19 35 18" src="https://user-images.githubusercontent.com/35312127/103567419-1488d380-4ec4-11eb-81c3-888c9f69fc25.png">

**Comment by rern** [comment]
Try:
```sh
mpc -f '%album%^^[%albumartist%|%artist%]^^%file%' listall \
| awk -F'/[^/]*$' 'NF && !/^\^/ && !a[$0]++ {print $1}' \
| sort -u
```
And post only errors.

**Reply by pdo59** [reply]
Hi,
I let rAudio work tonight to update my entire collection. I rebuilt the entire database and this morning everything is ok. 
Also your mpc command has ran without errors. I don't know why, but now the library is ok.

<img width="1146" alt="Schermata 2021-01-05 alle 08 40 04" src="https://user-images.githubusercontent.com/35312127/103619220-9e22ba80-4f31-11eb-84f7-85d3971a1f1b.png">

**Comment by jelinj8** [comment]
Hello,
mine is even a bit stranger (with a rather big collection, by the song count a lot of scanning is still missing) - mark the missing album count:
![image](https://user-images.githubusercontent.com/3816495/104450760-e16bd180-55a0-11eb-8c0a-61794f57e120.png)
![image](https://user-images.githubusercontent.com/3816495/104450798-f0eb1a80-55a0-11eb-9bf2-2099e6a35f3b.png)

MPD update of library ended (after some hours of scanning) with this:
```
...
Jan 13 12:43:12 rAudio taskset[403]: update: added NAS/Music/Roxette/Roxette_Room Service_/Roxette/It Takes You No Time To Get Here.mp3
Jan 13 12:43:12 rAudio taskset[403]: update: added NAS/Music/Roxette/Roxette_Room Service_/Roxette/Looking For Jane.mp3
Jan 13 12:43:12 rAudio taskset[403]: update: added NAS/Music/Roxette/Roxette_Room Service_/Roxette/Fool.mp3
Jan 13 12:43:12 rAudio taskset[403]: update: added NAS/Music/Roxette/Roxette_Room Service_/Roxette/Make My Head Go Pop.mp3
Jan 13 12:43:17 rAudio taskset[403]: exception: error on client 19502: Output buffer is full
Jan 13 12:43:17 rAudio mpdidle.sh[23657]: MPD error: Connection closed by the server
Jan 13 12:43:52 rAudio taskset[403]: exception: No such directory
Jan 13 13:01:01 rAudio CROND[27273]: (root) CMD (run-parts /etc/cron.hourly)

```
Maybe some buffer is a problem when scanning a large library? Anyway even additional scan of library (be it rebuild or just a modified scan) didn't fix the empty "albums" category.
The same library scans OK with MPD itself (MoOde audio, or a standalone MPD on a NanoPi NEO2).

I've tried to scan just a USB flash with two albums and that worked. It failed on an attempt to play one of them (probably some dangerous value in an id3tag?)

**Comment by pdo59** [comment]
Hello,
in my previous I wrote that rebuilding my entire collection I had corrected the "Albums" view.  Unfortunately, this was not the case. After some upgrades to the collection (which weren't visible in the "Albums" view anyway) I re-launched a full update and now I can't find anything in "Albums" again.

<img width="1050" alt="Schermata 2021-01-13 alle 17 52 31" src="https://user-images.githubusercontent.com/35312127/104483426-65857f80-55c8-11eb-9651-c0eb275b4978.png">

**Comment by pdo59** [comment]
Hi all,
ok, I discovered the problem, in my collection there are some album with the character **`"`** in  the title. 
This prevents the `cmd-list.sh` script from finalizing the creation of the `/srv/http/data/mpd/album` and `/srv/http/data/mpd/albumbyartist` files. Currently I have edited album titles but it would be useful to edit the script.
Another thing: I also changed the value of `max_output_buffer_size` to `"65536"` to avoid the mpd message `MPD error: Connection closed by the server`

<img width="893" alt="park" src="https://user-images.githubusercontent.com/35312127/104646429-afe22b80-56b0-11eb-8bfe-665d8c333253.png">
